helm/templates/cluster-role-controller.yaml

```diff
@@ -46,6 +46,15 @@ rules:
   - list
   - patch
   - watch
+- apiGroups:
+  - coordination.k8s.io
+  resources:
+  - leases
+  verbs:
+  - create
+  - get
+  - list
+  - update
```

Comment:
those already exist in https://github.com/aws-controllers-k8s/s3-controller/blob/main/helm/templates/leader-election-role.yaml#L12-L23 . Did you try to install the controller with helm value `leaderElection.enabled=true` ?

Reply:
@binchenX please file a github issue if you think the helm installation is broken with leaderElection flag

Reply:
indeed there seems to be a discrepancy between:
    [- "$(ENABLE_LEADER_ELECTION)"](https://github.com/aws-controllers-k8s/s3-controller/blob/main/helm/templates/deployment.yaml#L62)
and
    [- name: ENABLED_LEADER_ELECTION](https://github.com/aws-controllers-k8s/s3-controller/blob/main/helm/templates/deployment.yaml#L94)

Reply:
indeed https://github.com/aws-controllers-k8s/code-generator/pull/466/files